swarm/network/stream/syncer.go

```diff
@@ -69,76 +68,103 @@ func (s *SwarmSyncerServer) Close() {
 
 // GetData retrieves the actual chunk from netstore
 func (s *SwarmSyncerServer) GetData(ctx context.Context, key []byte) ([]byte, error) {
-	chunk, err := s.store.Get(ctx, storage.Address(key))
+	ch, err := s.store.Get(ctx, chunk.ModeGetSync, storage.Address(key))
 	if err != nil {
 		return nil, err
 	}
-	return chunk.Data(), nil
+	return ch.Data(), nil
 }
 
 // SessionIndex returns current storage bin (po) index.
 func (s *SwarmSyncerServer) SessionIndex() (uint64, error) {
-	return s.store.BinIndex(s.po), nil
+	return s.store.LastPullSubscriptionBinID(s.po)
 }
 
-// GetBatch retrieves the next batch of hashes from the dbstore
+// SetNextBatch retrieves the next batch of hashes from the localstore.
+// It expects a range of bin IDs, both ends inclusive in syncing, and returns
+// concatenated byte slice of chunk addresses and bin IDs of the first and
+// the last one in that slice. The batch may have up to BatchSize number of
+// chunk addresses. If at least one chunk is added to the batch and no new chunks
+// are added in batchTimeout period, the batch will be returned. This function
+// will block until new chunks are received from localstore pull subscription.
```

Comment:
Is `StartNextBatch` a more descriptive name for this method?

> and returns concatenated byte slice of chunk addresses and bin IDs of the first and the last one in that slice [...] The batch may have up to BatchSize number of chunk addresses

Ambiguous; sounds like it contains _only_ first and last addrs _and_ up to batchsize addrs at the same time.

> If at least one chunk is added to the batch and no new chunks are added in batchTimeout period, the batch will be returned. This function will block until new chunks are received from localstore pull subscription.

This is not very well written. How about something like:

```
This function receives chunks from the localstore pull subscription. It will return when one of the following conditions are met:
* "maxBatchSize" chunks have been received
* First chunk has been received **AND** last chunk was received longer than `batchTimeout` ago
```